transformer_engine/common/include/transformer_engine/softmax.h

```diff
@@ -15,23 +15,45 @@
 extern "C" {
 #endif
 
-
+/*! \brief Compute scaled softmax activation on the input.
+ *
+ *  \param[in]     input           Input tensor for softmax.
+ *  \param[out]    softmax_results Output tensor.
+ *  \param[in]     scale_factor    Scalar for the input tensor.
+ *  \param[in]     stream          CUDA stream used for the operation.
+ */
 void nvte_scaled_softmax_forward(
     const NVTETensor input,
     NVTETensor softmax_results,
     float scale_factor,
     cudaStream_t stream
 );
 
-
+/*! \brief Compute the backward of the scaled softmax activation.
+ *
+ *  - `output_grads` is the input tensor containing the gradients received from the following layer.
+ *  - `softmax_results` is the output tensor containing the computed gradients.
+ *
+ *  \param[in]     output_grads    Input tensor for softmax.
+ *  \param[out]    softmax_results Output tensor.
+ *  \param[in]     scale_factor    Scalar for the output tensor.
+ *  \param[in]     stream          CUDA stream used for the operation.
+ */
 void nvte_scaled_softmax_backward(
     const NVTETensor output_grads,
-    const NVTETensor softmax_results,
```

Comment:
Based on the previous comment I believe this change is wrong and it is the `output_grads` that is the output (which actually begs the question where is the incoming gradient - is it doing computation in place for some reason?).

Reply:
Actually yes you are right, the kernels take two separate arguments, `gradInput` and `grad`, the latter of which is supposed to be the incoming gradients. But those `dispatch_*` function pass `output_grads.dptr` to both these arguments effectively making an in place op.

Reply:
We should separate them then in the API and leave it to the user to decide whether they want in place or out of place.